### ISSUE TITLE
Unexpected variance when using batchnorm

### ISSUE THREAD

**Issue by maxbry** [keras-team/keras#8293]
I've taken the mnist ccn example and made some minor changes to it by adding batch norm and changing the activation to tanh:

    model = Sequential()
    model.add(Conv2D(32, kernel_size=(3, 3), input_shape=input_shape))
    model.add(BatchNormalization())
    model.add(Activation('tanh'))
    
    model.add(Conv2D(64, (3, 3)))
    model.add(BatchNormalization())
    model.add(Activation('tanh'))

    model.add(MaxPooling2D(pool_size=(2, 2)))
    model.add(Dropout(0.25))
    model.add(Flatten())
    
    model.add(Dense(128))
    model.add(BatchNormalization())
    model.add(Activation('tanh'))

    model.add(Dropout(0.5))
    model.add(Dense(num_classes, activation='softmax'))

After successfully training the network, I'm printing information about the layers, meaning the `min`, `max`, `avg` and `std` of every layer:

    for layer in model.layers:
        print(layer.name)

        for w in layer.get_weights():
            w_min = w.min()
            w_max = w.max()
            w_avg = w.mean()
            w_std = w.std()
      
            print('%.4f %.4f %.4f %.4f' % (w_min, w_max, w_avg, w_std))

I'm getting following output:

    conv2d_1
    -0.2803 0.2530 0.0008 0.1156
    -0.0847 0.1228 -0.0021 0.0496
    batch_normalization_1
    0.9764 1.4356 1.1471 0.1074
    -0.4299 0.4199 -0.0062 0.3002
    -0.1278 0.0862 -0.0008 0.0463
    0.0018 0.0287 0.0090 0.0065
    activation_1
    conv2d_2
    -0.2814 0.3204 -0.0001 0.0788
    -0.1228 0.1248 0.0069 0.0556
    batch_normalization_2
    0.7068 1.3278 1.0482 0.1544
    -0.5822 0.2336 -0.1585 0.1900
    -1.2246 0.7163 -0.1448 0.4045
    0.5834 2.8951 1.2963 0.5140
    activation_2
    max_pooling2d_1
    dropout_1
    flatten_1
    dense_1
    -0.2887 0.2982 -0.0004 0.0521
    -0.0008 0.0009 -0.0000 0.0003
    batch_normalization_3
    1.1302 1.5769 1.3250 0.1044
    -0.2698 0.2312 -0.0071 0.0989
    -25.3391 21.4002 -0.9373 11.4734
    68.2413 258.5824 136.4436 34.9851
    activation_3
    dropout_2
    dense_2
    -0.3922 0.3955 -0.0010 0.2013
    -0.2217 0.2049 -0.0148 0.1292

My question now is: How can it be that the running mean and running variance of that one batchnorm layer are that high? As I see it, the incoming values are roughly `[-1, 1]` because of tanh and because the quite small weights of the conv layers.

**Comment by maxbry** [comment]
Bump
